### ISSUE TITLE
Create a dashboard to show our team's "time off" progress

### ISSUE THREAD

**Issue by choldgraf** [2i2c-org/team-compass#592]
### Context

We currently recommend that team members take **at least** 40 days of time off each year. This includes holidays, personal time off, etc. There's no hard limit, and we anticipate folks will take off in the ballpark of 40 days. However, we much prefer that people _overshoot_ than _undershoot_ time off. Many organizations with "no limit to PTO" policies end up with _less time off_ than more. How can we avoid this?

### Proposal

I propose that we create a lightweight dashboard that is available only to our team (e.g. `2i2c.org/team-dashboard/time-off`) and that visualizes our "time off progress" throughout the year. Basically, we want a quick way to visualize whether people are over- or under-shooting 40 days a year.

As an example, here's a quick viz I put together from our current Google Calendar entries for 2022:

![image](https://user-images.githubusercontent.com/1839645/207833373-e606eadf-a173-4797-ae3b-40b3cfe0c823.png)

**Comment by sgibson91** [comment]
My concern is that taking time off can be quite personal, and maybe sharing this openly across the whole team isn't the best option. I do see a benefit for line managers having this information and having private discussions with their reports about it to ensure wellbeing etc. But then we end up in the rabbit hole of "but we don't have a formal line management system"...

I don't think this is _particularly_ about stress or inequity, but more about over-sharing information that isn't required to be shared. Maybe it's a webapp that you log into and only pulls _your_ data?

**Comment by GeorgianaElena** [comment]
Curios if this dashboard take into account `reduced availabilities` for when people went to conferences, as I really don't think that qualifies exactly as time off? Also, there might also be issues like https://github.com/2i2c-org/meta/issues/422, when the time off was canceled.

> Would a dashboard like this be a potential source of stress or inequity with the team? The goal is to encourage ourselves to take time off, but if it is a source of stress then I wouldn't want to post it.

This board may or may not be public for the team. Personally, it doesn't mind me to have it shared with the team, but I would feel more cautious about sharing it with the world. At least not share it in this format, where someone can understand the time off pattern of someone (in this idea, I suggest moving this very issue to the meta repo).

But I believe, it should be analyzed at the end of year and taken into account. This can reveal things that are worth to be taken into consideration. Things like burn out, or even the need to grow the team more as people might feel less likely to take time off if they often have lots of urgent things they have to resolve (ref https://github.com/2i2c-org/meta/issues/357).

So, if we honestly want that people to take the min 40 days off, then we should try and make if happen.

**Comment by sgibson91** [comment]
> it doesn't mind me to have it shared with the team

Yeah, I think this issue isn't super big _right now_, but I can see it becoming a pain point as we scale and grow and the vibe is not so much "I get to work with my friends, yay!"

E.g., I kind of feel this way about the last part of the time-off template being "any team member can raise concerns". I think it's ok now because we all have high levels of trust in one another. But as the team grows we won't know each other as personally and one team member objecting to another team member's time off (which _could_ result in cancellation, it hasn't yet, but that's the implication of that policy) could cause grudges between team members.

**Comment by damianavila** [comment]
Several good points were raised by @GeorgianaElena and @sgibson91.

> My concern is that taking time off can be quite personal, and maybe sharing this openly across the whole team isn't the best option. 

I do agree this openness can be a problem.

> I do see a benefit for line managers having this information and having private discussions with their reports about it to ensure wellbeing etc.

I agree as well. It would be super useful for managers.

> Maybe it's a webapp that you log into and only pulls your data?

I think it is a good idea to have this information on a pull-based basis.

> Curios if this dashboard take into account reduced availabilities for when people went to conferences, as I really don't think that qualifies exactly as time off? 

Good point, I would also like to know if these calculations are including weekends... because I noticed when people add a long period of time in the shared calendar, they usually encompass the weekends that should NOT be accounted for.

>  would feel more cautious about sharing it with the world.

This should be definitely an internal tool!

> Things like burn out, or even the need to grow the team more

Indeed!!

> But as the team grows we won't know each other as personally and one team member objecting to another team member's time off (which could result in cancellation, it hasn't yet, but that's the implication of that policy) could cause grudges between team members.

I agree, the approach works for now but it does not scale and needs to be worked out in the future.

**Comment by choldgraf** [comment]
A few quick thoughts here:

- Yep definitely internal. This is why I was playing around with this PR :-) https://github.com/2i2c-org/team-compass/pull/585
- Agreed it wouldn't scale - I feel like it should be available at the manager level so that they can get a birds-eye view of their team's availability.
- Note that if we adopt a GitHub Project-based model for this, then a similar view will be available: https://github.com/orgs/2i2c-org/projects/39/views/4
- Agree we should exclude weekends - I will try and recalculate just for viz purposes.
- Agreed that this is only "vacation". It should not include "reduced", nor should it include "out because of a work trip". The point is to disconnect **entirely** and this is what we want to encourage and track.
- If we use a setup like [the project board](https://github.com/orgs/2i2c-org/projects/39/views/4), we could have a dropdown for "type" where people could mark "reduced" or "vacation". "reduced" assumes you're still working (so either a personal reduced day, or you're busy w/ other stuff like traveling **for work**)

**Comment by choldgraf** [comment]
Well that explains why Claire said it didn't feel like I had taken the most vacation of everybody :-D 

After re-calculating with **only** the "AWAY" events (not "REDUCED"), and excluding weekends, the leaderboard changes!

![image](https://user-images.githubusercontent.com/1839645/208076633-ec89f774-5dd2-481a-aed6-a5f478d9e21e.png)

I am now in last place of our full-time staff that worked all year in 2022 😭 .

BTW, moving this to meta because as others rightly pointed out, this should be private to the team.
